### PR TITLE
Quarto api interlinks

### PR DESCRIPTION
**Pull request by machow**
This PR addresses parts of https://github.com/ibis-project/ibis/issues/6814, by 

- [x] : adding the quartodoc interlinks filter
- [x] : shortening method names (e.g. from `expr.types.generic.Column.lead(...)` to `lead(...)`
- [x] : fixing code highlighting

Note that as part of this, I updated quartodoc, but it seems like something about that screwed up the poetry.lockfile. It seems to have taken out the category fields :/.

@cpcloud, I wasn't able to find in the CI where docs are generated. Do you mind pointing me in the right direction? I also have never run justfiles, so crammed the commands in there for now, and will take a crack at making sure they work from there a sec!

### CODE REVIEW

**Comment by machow** [comment]
Whoops, I've now installed pre-commit, but I'm getting a surprising error about prettier not being installed. Any idea what's needed here?

image

**Comment by cpcloud** [comment]
> Do you mind pointing me in the right direction?

For PRs we build the docs with these two steps: https://github.com/ibis-project/ibis/blob/quarto/.github/workflows/ibis-docs-lint.yml#L176-L180

After a merge to `master` there's one more step that deploys the rendered docs to netlify using `just docs-deploy`: https://github.com/ibis-project/ibis/blob/quarto/.github/workflows/ibis-docs-lint.yml#L226-L229

> so crammed the commands in there for now

Totally fine!

File: docs/_renderer.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import griffe.expressions as expr  # noqa: F401
+import quartodoc.ast as qast
+from plum import dispatch
+from quartodoc import MdRenderer
+
+
+class Renderer(MdRenderer):
+    style = "ibis"
+
+    @dispatch
```
Comment (cpcloud) [inline docs/_renderer.py:14]: image

**Comment by cpcloud** [comment]
@machow I can help with the failing generated files check.

**Comment by cpcloud** [comment]
I can help with the other failure too!

**Comment by cpcloud** [comment]
@machow I tried this out locally and the `_renderer.py` is working!

Is this ready for merge?

File: justfile [hunk 2]
```diff
@@ -136,6 +136,8 @@ profile +args:
     pyinstrument {{ args }}
 
 docs-apigen *args:
+    cd docs && quarto add machow/quartodoc --no-prompt
```
Comment (cpcloud) [inline justfile:139]: I suppose we can remove this line now since the extension is checked into the repo, but I can do that in a follow-up.

**Comment by cpcloud** [comment]
Ah I see `shortening method names` is still unchecked!

**Comment by machow** [comment]
Alright, I think it should be working! (Happy to tweak!)

**Review by cpcloud — APPROVED** [review]
LGTM! Thanks @machow for all your work on this and `quartodoc` 🚀

**Pull request merged**